src/unix/tty.c

```diff
@@ -293,7 +293,14 @@ uv_handle_type uv_guess_handle(uv_file file) {
   if (file < 0)
     return UV_UNKNOWN_HANDLE;
 
+#if defined(__PASE__)
+  /* On IBMi PASE isatty() always returns true for stdin, stdout and stderr.
+   * Use ioctl() instead to identify whether it's actually a TTY.
+   */
+  if (!ioctl(file, (TXISATTY + 0x81), NULL))
```

Comment:
```suggestion
  if (!ioctl(file, TXISATTY + 0x81, NULL))
```
Would it make sense to check that `errno == ENOTTY` when ioctl() returns -1? The fstat() call below catches cases like EBADF but are there other errors that may need to be handled here?

Reply:
I will try to make it clear.

Reply:
I tried below code in a TN5250 terminal. The errno is ENOTTY, which is expected.
```
#include <sys/ioctl.h>
#include <errno.h>
#include <stdio.h>

int main(){
    int r = ioctl(0, (TXISATTY + 0x81), NULL);
    printf("ioctl returns %d, errno is %d\n", r, errno);
}
```
```
> ./isatty                     
  ioctl returns -1, errno is 25
  $                            
```
```
$ cat /usr/include/errno.h | grep ENOTTY
#define ENOTTY  25      /* Inappropriate I/O control operation  */
```

Reply:
- If `ioctl(file, TXISATTY + 0x81, NULL)` returns `0`, it means this is a TTY, then the `if(!ioctl...)` equals `if(isatty...)`.
- If `ioctl(file, TXISATTY + 0x81, NULL)` returns `-1` and the errno is `ENOTTY`, then the `if(!ioctl...)` equals `if(!isatty...)`.

Shall I update it to this? --
```
if (!ioctl(file, (TXISATTY + 0x81), NULL) || errno != ENOTTY)
```

Reply:
The PASE `isatty` code maps all `errno`s except `EBADF` to `ENOTTY`. If anything, we should mimic that.

This could be a lot cleaner if we extracted the `ioctl` code and moved it to a separate function like `isatty_pase` then `#define isatty isatty_pase`.